follej/session03/mailroom.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+from statistics import mean
+
+donors = [[]] * 5
+
+donors[0] = ['William Gates, III', [65000, 68888.40]]
+donors[1] = ['Mark Zuckerberg', [65000, 68888.40, 600000]]
+donors[2] = ['Jeff Bezos', [65000, 68888.40, 10000]]
+donors[3] = ['Paul Allen', [65000, 68888.40]]
+donors[4] = ['Jesse Follet', [0.65, 1.50]]
+
+
+def create_a_report():
+    indices = sort_donors()
+    print_a_report(indices)
+
+
+def print_a_report(indices):
+    header = 'Donor Name                | Total Given | Num Gifts | Average Gift'
+    top_line = '-' * len(header)
+    print(header)
+    print(top_line)
+    for i in indices:
+        print(donors[i][0].ljust(26), end='')
+        print("$%12.2f" % (sum(donors[i][1])), end='')
+        print("%12d  " % (len(donors[i][1])), end='')
+        print("$%12.2f" % (mean(donors[i][1])))
+    print()
+
+
+def sort_donors():
+    amounts = []
+    for i in range(len(donors)):
+        amounts.append([sum(donors[i][1]), i])
+    sort_amounts = sorted(amounts, reverse=True)
+    sort_indices = []
+    for i in range(len(sort_amounts)):
+        sort_indices.append(sort_amounts[i][1])
+    return sort_indices
+
+
+def quit_app():
+    print("Quitting!")
+    quit()
+
+
+def init_menu():
+    response = 0
+    while response != 3:
+        print("1) Sent a Thank You")
+        print("2) Create a Report")
+        print("3) Quit")
+        response = int(input("> "))
+        print(response)
+        if response == 1:
+            send_a_thank_you()
+        elif response == 2:
+            create_a_report()
+        else:
+            pass
+
+
+def write_thank_you_email(donor_index):
+    print("To {}:".format(donors[donor_index][0]))
+    print()
+    print("Thank you for you generous donation of $%.2f" % sum(donors[donor_index][1]))
+
+
+def get_donor_names_list():
+    donor_names = []
+    for i in range(len(donors)):
+        donor_names.append(donors[i][0])
+    return donor_names
+
+
+def send_a_thank_you():
+    donor_names = get_donor_names_list()
+    response = "list"
+    while response != "":
```

Comment:
Not sure you need a while loop here.  You have only 3 possible answers, and no real bad input value to validate. `Enter` to return to main menu, a name in the list, and name not in the list.  If you wanted to start the loop over again you can have your final `else` just call the same function again, or call the main menu.